mm2src/mm2_main/src/rpc/lp_commands/lr_swap/lr_impl.rs

```diff
@@ -266,17 +254,14 @@ impl LrDataMap {
                 .with_include_tokens_info(Some(true))
                 .with_include_gas(Some(true))
                 .build_query_params()?;
-            let url_builder = SwapUrlBuilder::create_api_url_builder(ctx, chain_id, SwapApiMethods::ClassicSwapQuote)?;
-            let fut = ApiClient::call_one_inch_api::<ClassicSwapData>(url_builder, Some(query_params));
+            let url = SwapUrlBuilder::create_api_url_builder(ctx, chain_id, SwapApiMethods::ClassicSwapQuote)?
+                .with_query_params(query_params)
+                .build()?;
+            let fut = ApiClient::call_api::<ClassicSwapData>(url);
             quote_futs.push(fut);
             src_dst.push((src_token.clone(), dst_token.clone()));
         }
-
-        let swap_data = join_all(quote_futs)
-            .await
-            .into_iter()
-            .filter_map(|res| if let Ok(lr_data) = res { Some(lr_data) } else { None }) // skip if bad result received (for e.g. low liguidity)
-            .collect::<Vec<_>>();
+        let swap_data = join_all(quote_futs).await.into_iter().map(|res| res.ok()); // if a bad result received (for e.g. low liguidity) set to None to preserve swap_data length
         let swap_data_map = src_dst.into_iter().zip(swap_data.into_iter()).collect();
```

Comment:
nit: i think the .into_iter() in swap_data.into_iter() could be removed here. since it's already an iter.